### PR TITLE
 #3074 #2765 Applying CS and adding lacking types wherever holes have been forgotten

### PR DESCRIPTION
**Pull request by Ocramius**
Applies to doctrine/dbal#2765 and doctrine/dbal#3074

### CODE REVIEW

**Comment by morozov** [comment]
@Ocramius sorry, didn't notice your PR and fixed CS violations by rebasing. Do you want to rebase on top and keep your type changes? The code style build is green now.

**Comment by Ocramius** [comment]
@morozov can do once at the office

**Comment by morozov** [comment]
Feel free to push into my branch. I'm tired restarting errored MySQL builds on Travis :-\

File: lib/Doctrine/DBAL/Driver/PDOConnection.php [hunk 1]
```diff
@@ -183,7 +183,7 @@ public function trackLastInsertId() : void
      *
      * @return string The last insert ID or '0' in case the last insert ID generated on this connection is unknown.
      *
-     * @throws PDOException
+     * @throws \PDOException
```
Comment (Ocramius) [inline lib/Doctrine/DBAL/Driver/PDOConnection.php:186]: Exception type changed to base exception here, since the wrapped connection is accessed directly.

**Comment by Ocramius** [comment]
@morozov rebased :+1:

File: tests/Doctrine/Tests/DBAL/Functional/LastInsertIdTest.php [hunk 2]
```diff
@@ -13,7 +13,7 @@ class LastInsertIdTest extends DbalFunctionalTestCase
     /** @var Connection */
     private $testConnection;
 
-    protected function setUp()
+    protected function setUp() : void
```
Comment (morozov) [inline tests/Doctrine/Tests/DBAL/Functional/LastInsertIdTest.php:16]: Should we enforce this in the code style? Or maybe recommend to PHPUnit? Seems working even on PHP 7.2: https://3v4l.org/4QqdS

File: tests/Doctrine/Tests/DBAL/Functional/LastInsertIdTest.php [hunk 3]
```diff
@@ -47,27 +47,27 @@ private function createTable($tableName)
         $connection->getSchemaManager()->createTable($table);
     }
 
-    public function testLastInsertIdNoInsert()
+    public function testLastInsertIdNoInsert() : void
```
Comment (morozov) [inline tests/Doctrine/Tests/DBAL/Functional/LastInsertIdTest.php:50]: This also could be enforced in the code style.

**Pull request merged**